pkgs/by-name/xr/xrootd/package.nix

```diff
@@ -0,0 +1,160 @@
+{
+  lib,
+  stdenv,
+  callPackage,
+  fetchFromGitHub,
+  davix,
+  cmake,
+  gtest,
+  makeWrapper,
+  pkg-config,
+  curl,
+  isa-l,
+  fuse,
+  libkrb5,
+  libuuid,
+  libxcrypt,
+  libxml2,
+  openssl,
+  readline,
+  scitokens-cpp,
+  systemd,
+  voms,
+  zlib,
+  # If not null, the builder will
```

Comment:
Why do we need this? Sounds a bit odd to me

Reply:
The comment is no longer relevant after #241466. The current version would be "If not null, move the default configuration files to `"$etc/etc"` and look for the configuration directory at `externalEtc`. Otherwise, the program will look for the configuration files under `$out/etc`."

I made `xrootd` reference configurations inside `/etc` (or anywhere specified by `externalEtc`) instead of `$out/etc` by default, as it is sensible to place the configuration files in a separate directory instead of rebuilding the package every time the configuration changes. This also makes `xrootd` work out-of-the-box (also) for the FHS environment.

An alternative implementation method is to set `-DCMAKE_INSTALL_DATADIR` directly to `/etc` (or anywhere specified by `externalEtc`). This works only when the package manager installs the default configuration files and not the upstream build system and would fail when the XRootD (upstream) people decide to generate the configuration files using CMake someday, as some other packages like `xrootd` do. The current implementation method (placing a symbolic link inside `$out/etc` pointing to the path specified by `externalEtc`, and placing the default configuration files inside `$etc/etc/xrootd`)

Reply:
> The current version would be "If not null, move the default configuration files to `"$etc/etc"` and look for the configuration directory at `externalEtc`. Otherwise, the program will look for the configuration files under `$out/etc`."

I updated the comment.